game/scripts/vscripts/items/shroud.lua

```diff
@@ -0,0 +1,84 @@
+LinkLuaModifier( "modifier_ghost_state", LUA_MODIFIER_MOTION_NONE )
+LinkLuaModifier( "modifier_item_glimmer_cape_fade", LUA_MODIFIER_MOTION_NONE )
+LinkLuaModifier( "modifier_item_shroud_passive", "items/shroud.lua", LUA_MODIFIER_MOTION_NONE )
+
+--------------------------------------------------------------------------------
+
+item_shroud = class({})
+
+function item_shroud:GetIntrinsicModifierName()
+	return "modifier_item_shroud_passive"
+end
+
+function item_shroud:OnSpellStart()
+	local hTarget = self:GetCursorTarget()
+	local shroud_duration = self:GetSpecialValueFor( "duration" )
+
+	EmitSoundOn( "Item.GlimmerCape.Activate", hTarget )
+	hTarget:AddNewModifier( hTarget, self, "modifier_ghost_state", { duration = shroud_duration } )
+	hTarget:AddNewModifier( hTarget, self, "modifier_item_glimmer_cape_fade", { duration = shroud_duration } )
+end
+
+--------------------------------------------------------------------------------
+
+item_shroud_2 = class({})

```

Comment:
Yeah, it's super useful for our upgrading items since the tiers always function the same.